### DISCUSSION TITLE
How to Troubleshoot UI connection to Edge Devices?

### DISCUSSION THREAD

**Discussion by jjsenay** [ambianic/ambianic-edge#409] [Q&A] [answered]
Do to network changes on the Internet Backbone, a TURN Server needs to be integrated into the Ambianic Edge for connection between the Edge Server and the UI-PWA.

**Comment by jjsenay** [comment]
On Friday, December 10 my connections to (3) local Edge devices stopped working in the UI-PWA.  Below are the Screen and Log captures of the events.

"All, Having a problem with my Edge devices.  I know the Edge device in my garage is working, because I keep getting email notifications from the IFTTT api.  The browser is Safari.  I ran it several times to show the java console from the google email before I recorded it."

https://user-images.githubusercontent.com/8701163/146043373-61971cbd-5e53-495b-963b-b950e954eaf0.mov

All 3 of my Edge Devices show up in the ambianic log console

When I get to the UI-PWA error I try then to connect the Edge to the network.  Here is the video and log file.

https://user-images.githubusercontent.com/8701163/146044110-a302bce6-ef46-4a29-8b3e-235165867f94.mov

[UI-PWA-reconnet.txt](https://github.com/ambianic/ambianic-edge/files/7713401/UI-PWA-reconnet.txt)

Ivelin Ivanov  11:23 AM
You are most likely behind a symmetric NAT. https://webrtchacks.com/symmetric-nat/
webrtcHackswebrtcHacks
Am I behind a Symmetric NAT? - webrtcHacks
Network Address Translation (NAT) can be a nuisance for VoIP. Fippo gives background and code for identifying the Symmetric NAT scenario in WebRTC.
May 15th, 2017
11:24
I am working on a patch that should work around that without you having to understand your router configuration and ISP side settings.

John Senay  11:28 AM
If I take (2) of the Edges off line, should the 3rd Edge "work" in my symmetric NAT Environment?

Mike Chapman  11:33 AM
I think it’s happening to me as well.

John Senay  11:40 AM
Powered down (2) of the Edges.  Still getting the same results.  Behaves the same with the VPN turned on or off

Ivelin Ivanov  9:19 PM  Dec 13,
Can you create a discussion on github so we can put all the materials in one place. https://github.com/ambianic/ambianic-edge/discussions
9:20
I am pushing out a new version. Found out that a bunch of providers pushed out updates after the big AWS outage.
9:21
Apple also pushed out iOS update that screws with WebRTC. Looks like a lot of companies panic locked a bunch of networking traffic.

**Comment by ivelin** [comment, marked as answer]
I did a lot of testing with various settings and seem to have found a configuration that works. Patches were pushed for both UI and Edge with WebRTC routing through TURN servers.

Note that Turn services incur hosting costs. We are currently on Free entry level development plans that are shared among all Ambianic users and devices. If we exceed this plans, we will need to upgrade to paid and hence find ways to fund these costs.

Fortunately STUN is sufficient for 80% of cases. We are also being extra careful in the product design to minimize bandwidth consumption for optimal user experience. Low bandwidth also helps with minimizing TURN traffic.

Please refresh your PWA to get the latest version and let me know your connectivity issues are now resolved.

**Reply by jjsenay** [reply]
Had the same  problem.  Updated the UI and Edge but still the same problem on the MAC with Brave.

https://user-images.githubusercontent.com/8701163/146057217-2b264003-046e-4f04-afb0-7ae0dd743662.mov

**Reply by ivelin** [reply]
The video you posted above shows that your PWA cannot discover an Ambianic Edge device on the local network. 

Are you sure both your laptop and the edge device are connected to the same WiFi router and use IP address assigned by that router?

Most likely they are on different private networks which don't share the same public IP address. 
A VPN client on your PWA device might be also causing this, even if the WiFi router is shared.
[This document](https://docs.ambianic.ai/users/devicemanagement/#local-discovery) details local connectivity mechanics in more detail.


If you know the Peer ID of each edge devices, you can skip local discovery and connect by using the Remote option on the Add Ambianic Edge device page. More details [here](https://docs.ambianic.ai/users/devicemanagement/#direct-peer-id-connection).

**Reply by jjsenay** [reply]
Let me update my VPN package and a monitoring package on my MAC, so just hold off for the time being

**Comment by jjsenay** [comment]
All,  cleaned up my MAC.  I deinstalled the UI-PWA and reinstalled.
Turned off all VPN and monitoring software.
Started the UI-PWA app.  connected with my Garage Edge, pulled a time line.
Went to add a new device on the local network and the UI showed the Office Edge and Great Room Edge.
Selected the Great Room Edge and I cannot pull a time line.
Attached is the JC log.
[Uploading ui.ambianic.ai-Great_Room.log…]()

**Reply by ivelin** [reply]
Can you upload another video with the dev console open so I can understand better what works and what doesn't.

**Comment by mchapman67** [comment]
Test reply

**Comment by mchapman67** [comment]
I’m willing to help with funds.

**Comment by jjsenay** [comment]
All,

Tested my Android PIXEL 2, turned off using  data, had to use WiFi with the UI-PWA and it is working.  I can connect with each Edge (3) and pull a time line from each Edge.

**Comment by jjsenay** [comment]
I repaired my car in my driveway.  Lots of Falls detected.  Some were false.  Use for training the AI model
[20211214-153553.415494.zip](https://github.com/ambianic/ambianic-edge/files/7714924/20211214-153553.415494.zip)

Enjoy!!!

**Reply by ivelin** [reply]
> I repaired my car in my driveway. Lots of Falls detected. Some were false. Use for training the AI model [20211214-153553.415494.zip](https://github.com/ambianic/ambianic-edge/files/7714924/20211214-153553.415494.zip)
> 
> Enjoy!!!

This is not connectivity related. Let's move to a separate discussion: "Fall Detection Model Training"

**Comment by mchapman67** [comment]
I can’t connect to the ambianic-edge… I keep getting a blank field when
attempting to associate my WiFi network with the ambianic-edge… not sure
why.

Mike

**Reply by jjsenay** [reply]
DO YOU HAVE MORE THAN (1) EDGE on the WiFi network?  Look at your networking on the phone.  IS their a SAVED ambianic-edge network?  IF there is unplug the Edge you are trying to get online.  Delete the Ambianic-Edge network.  Then plug in the Edge and watch for the Ambianic Edge network to pop up.  Give it the SSD and password

**Comment by mchapman67** [comment]
So this morning I dug up an older IPhone to see how it behaved with my Edge
and Balena connect, and I was able to connect. I forgot to check the IOS
version, but I do know IOS 15.1.1 seems to be really flaky (my connection
keeps dropping out). I had to leave for my office so I wasn’t able to
monitor the older IPhone.

In addition, I’m still experiencing issues connecting with the STUN server.
I’ve rebooted my Edge several times now.

I will take another look at the Edge later on today.

**Reply by ivelin** [reply]
Great feedback, Mike. Thank you!

**Comment by mchapman67** [comment]
The connection to the STUN service has been resolved, I’m using UI App
version 2.22.5. With that said, I’m still working on the camera issue.

**Reply by ivelin** [reply]
Great. Thanks for sharing.
